services/github/github-gist-stars.service.js

```diff
@@ -0,0 +1,97 @@
+import gql from 'graphql-tag'
+import Joi from 'joi'
+import { metric } from '../text-formatters.js'
+import { NotFound } from '../index.js'
+import { GithubAuthV4Service } from './github-auth-service.js'
+import { documentation as commonDocumentation } from './github-helpers.js'
+
+const schema = Joi.object({
+  data: Joi.object({
+    viewer: Joi.object({
+      gist: Joi.object({
+        stargazerCount: Joi.number(),
+        url: Joi.string(),
+      }).allow(null),
+    }).required(),
+  }).required(),
+}).required()
+
+const documentation = `${commonDocumentation}
+<p>This badge shows the number of stargazers for a gist. Gist id is accepted as input and 'gist not found' is returned if the gist is not found for the given gist id.
+</p>`
+
+export default class GithubGistStars extends GithubAuthV4Service {
+  static category = 'social'
+
+  static route = {
+    base: 'github/stars/gists',
+    pattern: ':gist',
+  }
+
+  static examples = [
+    {
+      title: 'Github Gist stars',
+      namedParams: { gist: '47a4d00457a92aa426dbd48a18776322' },
+      staticPreview: {
+        label: this.defaultBadgeData.label,
+        message: metric(29),
+        style: 'social',
+      },
+      documentation,
+    },
+  ]
+
+  static defaultBadgeData = {
+    label: 'Stars',
+    color: 'blue',
+    namedLogo: 'github',
+  }
+
+  static render({ stargazerCount, url }) {
+    return { message: metric(stargazerCount), link: [url] }
```

Comment:
On the repo stars badge, we link the left side of the badge to the repo URL and the right side to `/stargazers`. Can we do the same thing for the gist version too please.
https://github.com/badges/shields/blob/8e1680d65d12b97356a587db929ecb80e874fd78/services/github/github-stars.service.js#L48-L51
Gists also have a similar stargazers URL e.g: https://gist.github.com/maratori/47a4d00457a92aa426dbd48a18776322/stargazers